src/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs

```diff
@@ -71,6 +71,17 @@ public override void OnCustom(string name, Activity activity, object payload)
             {
                 case EntityFrameworkCoreCommandCreated:
                     {
+                        if (this.options.SuppressDownstreamInstrumentation)
+                        {
+                            SuppressInstrumentationScope.Enter();
+
+                            // If we are suppressing downstream instrumentation then inject
+                            // context here. EFCore uses SqlClient, so
+                            // SuppressDownstreamInstrumentation means that the
+                            // OpenTelemetry instrumentation for SqlClient will not be
+                            // invoked.
+                        }
```

Comment:
This option should suppress SqlClient instrumentation

Reply:
unit test as this is new capability for this instrumentation?

Reply:
yes! The ElasticSearch one has one. Grpc.NET from main repo also has.